get_ci_vm/entrypoint.sh

```diff
@@ -176,6 +217,114 @@ has_valid_credentials() {
     return 0
 }
 
+has_valid_aws_credentials() {
+    req_env_vars AWSCLI EC2_SSH_KEY
+    local _awsoutput _cfgvar
+    _awsoutput=$($AWSCLI configure list 2>&1 || true)
+    dbg "$AWSCLI configure list"
+    dbg "$_awsoutput"
+    if egrep -qx 'The config profile.+could not be found'<<<"$_awsoutput"; then
+        dbg "AWS config/credentials are missing"
+        return 1
+    elif [[ ! -r "$EC2_SSH_KEY" ]] || [[ ! -r "${EC2_SSH_KEY}.pub" ]]; then
+        dbg "EC2 ssh keys are missing"
+        return 1
+    else
+        # It's possible the config. is incomplete, verify minimums
+        for _cfgvar in aws_access_key_id aws_secret_access_key region; do
+            # Detect both errors on missing profile and/or missing files.
+            if ! $AWSCLI configure get $_cfgvar &> /dev/null; then
```

Comment:
Is the difference between "missing" and "empty" important? I will assume so, but please remember that it may be important to you but not to anyone else, and when an end user sees this error, they will report it as "something about var", and they won't remember whether the magic word was "missing" or "empty", so it might be nice to provide a more verbose explanation such as "AWS variable '$_cfgvar' is not present in the configuration" and "AWS variable '$_cfgvar' is present in the configuration, but empty", along with perhaps a "WEIRD" or something indicating how likely/impossible this is to happen, and what to do about it.

Oh, also the duplicate call is unnecessary:
```bash
if ! foo=$(AWS...); then a; elif [[ -z "$foo" ]]; then b ...
```

Reply:
...Updated function to treat both unset and empty the same (`return 1`).